app/bundles/FormBundle/Controller/ResultController.php

```diff
@@ -392,6 +394,218 @@ public function deleteAction(Request $request)
+                $submission = $model->getEntity($id);
+                if (null === $submission) {
+                    continue;
+                }
+                if (!$this->security->hasEntityAccess('form:forms:editown', 'form:forms:editother', $submission->getCreatedBy())) {
+                    $flashes[] = $this->accessDenied(true);
+                    continue;
+                }
+                $domain = $this->getDomainFromSubmission($submission);
+                if ($domain) {
+                    $domainsToSave['*@'.$domain] = true;
+                }
+            }
+
+            if ($domainsToSave) {
+                $existing      = $this->coreParametersHelper->get('do_not_submit_emails', []);
+                $hasNewDomains = false;
+
+                foreach (array_keys($domainsToSave) as $d) {
+                    if (!in_array($d, $existing, true)) {
+                        $existing[]    = $d;
+                        $hasNewDomains = true;
+                    }
+                }
+                $configurator->mergeParameters(['do_not_submit_emails' => $existing]);
+                $configurator->write();
+
+                $this->enableDonotSubmitValidationOnUnconfiguredForms();
+
+                $flashes[] = [
+                    'type' => 'notice',
+                    'msg'  => 'mautic.form.result.markspam.batch.success',
+                ];
+            } else {
+                $flashes[] = [
+                    'type' => 'notice',
+                    'msg'  => 'mautic.form.result.markspam.batch.none',
+                ];
+            }
+        }
+
+        $viewParameters = [
+            'objectId' => $formId,
+            'page'     => $page,
+        ];
+
+        return $this->postActionRedirect(
+            [
+                'returnUrl'       => $this->generateUrl('mautic_form_results', $viewParameters),
+                'viewParameters'  => $viewParameters,
+                'contentTemplate' => 'Mautic\\FormBundle\\Controller\\ResultController::indexAction',
+                'passthroughVars' => [
+                    'mauticContent' => 'formresult',
+                ],
+                'flashes' => $flashes,
+            ]
+        );
+    }
+
+    private function getDomainFromSubmission(Submission $submission): ?string
+    {
+        $results = $submission->getResults();
+        $form    = $submission->getForm();
+        foreach ($form->getFields() as $field) {
+            if ('email' === $field->getType()) {
+                $alias = $field->getAlias();
+                if (!empty($results[$alias]) && str_contains($results[$alias], '@')) {
+                    return strtolower(substr(strrchr($results[$alias], '@'), 1));
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private function enableDonotSubmitValidationOnUnconfiguredForms(): void
+    {
+        /** @var FormModel $formModel */
+        $formModel  = $this->getModel('form.form');
```

Comment:
I know  Symfony won't fill in the dependency for a private method, but the right way how to do it is to send the service as a parameter and get it from the actual controller actions and from DI of those actions.
```suggestion
    private function enableDonotSubmitValidationOnUnconfiguredForms(FormModel $formModel): void
    {
```